clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -2501,7 +2501,8 @@ void CXXNameMangler::mangleQualifiers(Qualifiers Quals, const DependentAddressSp
     if (Context.getASTContext().addressSpaceMapManglingFor(AS)) {
       //  <target-addrspace> ::= "AS" <address-space-number>
       unsigned TargetAS = Context.getASTContext().getTargetAddressSpace(AS);
-      if (TargetAS != 0 || (Context.getASTContext().getLangOpts().SYCLIsDevice))
+      if (TargetAS != 0 ||
+          Context.getASTContext().getTargetAddressSpace(LangAS::Default) != 0)
```

Comment:
It looks like this change remove the need for SYCL bindings in `libclc` library.
Before this change, SYCL device compiler used non-standard mangling for NVPTX target and mangled differently raw pointer parameters and parameters annotated with `opencl_private` address space attribute, although there is no need to differentiate these cases.
New code says the different mangling is used for parameters annotated with `opencl_private` address space attribute only if they are mapped to non-0 address space, so SYCL mangling for NVPTX now aligned with OpenCL mangling for NVPTX and we be remove the hack from libclc CMake files.
@Naghasan, am I right?

Reply:
llvm-test-suite results seem to confirm that.
  Unsupported      : 236
  Passed           : 244
  Expectedly Failed:  25